src/Client/Connection.cpp

```diff
@@ -774,6 +775,8 @@ Packet Connection::receivePacket()
     {
         Packet res;
 
+        total_memory_tracker.checkMemoryUsage();
```

Comment:
Using prints i found out that both allocator and new/delete operations are called. But for some reason allocator don't error, and new/delete operations are using with flag not to throw error. So, this function helps to know that we've reached memory limit.